test/api/v3/integration/groups/POST-groups.test.js

```diff
@@ -0,0 +1,130 @@
+        })
+      )
+      .to.eventually.be.rejected.and.eql({
+        code: 401,
+        error: 'NotAuthorized',
+        message: t('messageInsufficientGems'),
+      });
+    });
+
+     context('public guild', () => {
+       it('creates a group', () => {
+         let groupName = "Test Public Guild";
+         let groupType = "guild";
+         let tmpUser;
+
+         return generateUser({balance: 1}).then((generatedUser) => {
+           tmpUser = generatedUser;
+           api = requester(tmpUser);
+           return api.post('/groups', {
+             name: groupName,
+             type: groupType
+           });
+         })
+         .then((result) => {
+           expect(result._id).to.exist;
+           expect(result.name).to.equal(groupName);
+           expect(result.type).to.equal(groupType);
+         })
+         .then(() => {
+           api = requester(tmpUser);
+         });
+       });
+     });
+
+     context('private guild', () => {
+       it('creates a group', () => {
+         let groupName = "Test Private Guild";
+         let groupType = "guild";
+         let groupPrivacy = "private";
+         let tmpUser;
+
+         return generateUser({balance: 1}).then((generatedUser) => {
+           tmpUser = generatedUser;
+           api = requester(tmpUser);
+           return api.post('/groups', {
+             name: groupName,
+             type: groupType,
+             privacy: groupPrivacy
+           });
+         })
+         .then((result) => {
+           expect(result._id).to.exist;
+           expect(result.name).to.equal(groupName);
+           expect(result.type).to.equal(groupType);
+           expect(result.privacy).to.equal(groupPrivacy);
+         })
+         .then(() => {
+           api = requester(tmpUser);
+         });
+       });
+     });
+   });
+
+   context('Parties', () => {
+     it('creates a party', () => {
+       let groupName = "Test Party";
+       let groupType = "party";
+
+       return api.post('/groups', {
+         name: groupName,
+         type: groupType
+       })
+       .then((result) => {
+         expect(result._id).to.exist;
+         expect(result.name).to.equal(groupName);
+         expect(result.type).to.equal(groupType);
+       })
+     });
+
+     it('prevents user in a party from creating a party', () => {
+        let tmpUser;
+        let groupName = "Test Party";
+        let groupType = "party";
+
+        return generateUser().then((generatedUser) => {
+          tmpUser = generatedUser;
+          api = requester(tmpUser);
+          return api.post('/groups', {
+            name: groupName,
+            type: groupType
+          });
+        })
+        .then(() => {
+          return expect(api.post('/groups')).to.eventually.be.rejected.and.eql({
+            code: 401,
+            error: 'NotAuthorized',
+            message: t('messageGroupAlreadyInParty'),
+          });
+        })
+        .then(() => {
+          api = requester(tmpUser);
```

Comment:
You should just change the initial before to a beforeEach so a new user gets created every time.